examples/nlp/semantic_similarity_with_keras_nlp.py

```diff
@@ -0,0 +1,292 @@
+improve it.
+
+Let's recompile our model with a different learning rate and see performance
+"""
+bert_classifier = keras_nlp.models.BertClassifier.from_preset(
+    "bert_tiny_en_uncased", num_classes=3
+)
+
+bert_classifier.compile(
+    loss=keras.losses.SparseCategoricalCrossentropy(from_logits=True),
+    optimizer=tf.keras.optimizers.Adam(1e-4),
+    metrics=["accuracy"],
+)
+
+bert_classifier.fit(train_ds, validation_data=val_ds, epochs=1)
+
+bert_classifier.evaluate(test_ds)
+"""
+This time we got ~72% validation accuracy on val and test split. Not bad for 1 epoch ! Let's save our model
+for now and learn how to perform inference with it. We took batch size of 512 to utilize our GPUs fully.
+
+Let's see if we can improve it further. Let's use a learning rate scheduler this time.
+"""
+
+
+class TriangularSchedule(keras.optimizers.schedules.LearningRateSchedule):
+    """Linear ramp up for `warmup` steps, then linear decay to zero at `total` steps."""
+
+    def __init__(self, rate, warmup, total):
+        self.rate = tf.cast(rate, dtype="float32")
+        self.warmup = tf.cast(warmup, dtype="float32")
+        self.total = tf.cast(total, dtype="float32")
+
+    def __call__(self, step):
+        step = tf.cast(step, dtype="float32")
+        multiplier = tf.cond(
+            step < self.warmup,
+            lambda: step / self.warmup,
+            lambda: (self.total - step) / (self.total - self.warmup),
+        )
+        return tf.maximum(self.rate * multiplier, 0.0)
+
+
+bert_classifier = keras_nlp.models.BertClassifier.from_preset(
+    "bert_tiny_en_uncased", num_classes=3
+)
+
+bert_classifier.compile(
+    loss=keras.losses.SparseCategoricalCrossentropy(from_logits=True),
+    optimizer=tf.keras.optimizers.experimental.AdamW(
+        TriangularSchedule(0.001, 1000, 5000)
+    ),
+    metrics=["accuracy"],
+)
+
+bert_classifier.fit(train_ds, validation_data=val_ds, epochs=3)
+
+"""
+With LR scheduler and AdamW optimizer we see that after completion of first epoch, 
+validation accuracy hikes upto ~77%, and upto ~80% in three epochs
+
+Let's evaluate our model on test set !
+"""
+
+bert_classifier.evaluate(test_ds)
+
+"""
+Our Tiny BERT achieved around ~85% of accuracy on test set with learning rate scheduler
+
+# Save and Reload the model
+"""
+bert_classifier.save("bert_classifier")
+restored_model = keras.models.load_model("bert_classifier")
+restored_model.evaluate(test_ds)
+
+"""
+# Inference
+
+Let's see how to perform inference with KerasNLP models
+"""
+
+# Convert to Hypothesis-Premise pair, for forward pass through model
+sample = (sample["hypothesis"], sample["premise"])
+sample
+
+"""
+Again, we don't need to tokenize the inputs explicitly as we are using the default preprocessor
+here
+"""
+predictions = bert_classifier.predict(sample)
+
+# Get the class predictions with maximum probabilities
+print(tf.math.argmax(predictions, axis=1).numpy())
```

Comment:
Should we show softmax rather than argmax here? So we show that these are probabilities coming out?